metrics/cer/cer.py

```diff
@@ -18,30 +18,50 @@
 
 import jiwer
 import jiwer.transforms as tr
+from packaging import version
 
 import datasets
+from datasets.config import PY_VERSION
 
 
-class SentencesToListOfCharacters(tr.AbstractTransform):
-    def process_string(self, s: str):
-        return list(s)
+if PY_VERSION < version.parse("3.8"):
+    import importlib_metadata
+else:
+    import importlib.metadata as importlib_metadata
 
-    def process_list(self, inp: List[str]):
-        chars = []
 
-        for sentence in inp:
-            chars.extend(self.process_string(sentence))
+SENTENCE_DELIMITER = ""
 
-        return chars
 
+if version.parse(importlib_metadata.version("jiwer")) < version.parse("2.3.0"):
 
-cer_transform = tr.Compose(
-    [
-        tr.RemoveMultipleSpaces(),
-        tr.Strip(),
-        SentencesToListOfCharacters(),
-    ]
-)
+    class SentencesToListOfCharacters(tr.AbstractTransform):
+        def __init__(self, sentence_delimiter: str = " "):
+            self.sentence_delimiter = sentence_delimiter
+
+        def process_string(self, s: str):
+            return list(s)
+
+        def process_list(self, inp: List[str]):
+            chars = []
+            for sent_idx, sentence in enumerate(inp):
+                chars.extend(self.process_string(sentence))
+                if self.sentence_delimiter is not None and self.sentence_delimiter != "" and sent_idx < len(inp) - 1:
+                    chars.append(self.sentence_delimiter)
+            return chars
+
+    cer_transform = tr.Compose(
+        [tr.RemoveMultipleSpaces(), tr.Strip(), SentencesToListOfCharacters(SENTENCE_DELIMITER)]
+    )
```

Comment:
why can't we use the old transform here ?
```python
cer_transform = tr.Compose(
    [
        tr.RemoveMultipleSpaces(),
        tr.Strip(),
        SentencesToListOfCharacters(),
    ]
)
```

Reply:
This transform in the current state does the same thing as the old one.

However, in `jiwer 2.3.0`, we can use `tr.ReduceToSingleSentence` in place of our transform, which by default uses a single space as a delimiter between sentences. And this change allows us to pass the same delimiter to both of them.

Additionally, we can easily change the sentence delimiter in the future this way (if we want to align with `jiwer` for instance).